app/models/pawn.rb

```diff
@@ -1,2 +1,25 @@
 class Pawn < Piece
+  def valid_move?(destination_x, destination_y)
+    x_difference = x_position - destination_x
+    y_difference = destination_y - y_position
+    return true if vertical_move(destination_y, x_difference, y_difference)
+    return true if horizontal_move(destination_x, x_difference, y_difference)
+    false
+  end
+
+  private
+
+  def vertical_move(destination_y, x_difference, y_difference)
+    return false unless y_difference == 1 && x_difference.zero?
+    return true unless vertical_obstruction?(destination_y, x_difference, y_difference)
+    false
+  end
+
+  # how to test for pawns first move??
```

Comment:
For first move, you know which row the piece should be in for starting position (based on color). So if it's not there, then can assume it's the first move.

Other option, could add a database column that counts the number of moves a piece has made, increment on each move (this is nicer solution, but takes a little more work).